setup.sh

```diff
@@ -0,0 +1,275 @@
+    opt="${OPTARG%%=*}"     # extract long option name
+    OPTARG="${OPTARG#$opt}" # extract long option argument (may be empty)
+    OPTARG="${OPTARG#=}"    # if long option argument, remove assigning `=`
+  fi
+  case $opt in
+  # note the leading colon
+  b | branch) BRANCH="$OPTARG" ;;
+  d | dir) DIR="$OPTARG" ;;
+  g | git-repo) GIT_REPO="$OPTARG" ;;
+  r | runpod) RUNPOD=true ;;
+  h) display_help && exit 0 ;;
+  *) display_help && exit 0 ;;
+  esac
+done
+shift $((OPTIND - 1))
+
+# This must be set after the getopts loop to account for $DIR changes.
+BASE_DIR="$(echo "$DIR" | cut -d "/" -f2)"
+VENV_DIR="$DIR/venv"
+
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+  # Check if root or sudo
+  root=true
+  if [ "$EUID" -ne 0 ]; then
+    root=false
+  fi
+
+  if env_var_exists RUNPOD_POD_ID || env_var_exists RUNPOD_API_KEY; then
+    RUNPOD=true
+  fi
+
+  env_var_exists() {
+    local env_var=
+    env_var=$(declare -p "$1")
+    if ! [[ -v $1 && $env_var =~ ^declare\ -x ]]; then
+      return 1
+    fi
+  }
+
+  get_distro_name() {
+    local line
+    if [ -f /etc/os-release ]; then
+      # We search for the line starting with ID=
+      # Then we remove the ID= prefix to get the name itself
+      line="$(grep -Ei '^ID=' /etc/os-release)"
+      line=${line##*=}
+      echo "$line"
+      return 0
+    elif command -v python >/dev/null; then
+      line="$(python -mplatform)"
+      echo "$line"
+      return 0
+    elif command -v python3 >/dev/null; then
+      line="$(python3 -mplatform)"
+      echo "$line"
+      return 0
+    else
+      line="None"
+      echo "$line"
+      return 1
+    fi
+  }
+
+  get_distro_family() {
+    local line
+    if [ -f /etc/os-release ]; then
+      # We search for the line starting with ID_LIKE=
+      # Then we remove the ID_LIKE= prefix to get the name itself
+      # This is the "type" of distro. For example, Ubuntu returns "debian".
+      if grep -Eiq '^ID_LIKE=' /etc/os-release >/dev/null; then
+        line="$(grep -Ei '^ID_LIKE=' /etc/os-release)"
+        line=${line##*=}
+        echo "$line"
+        return 0
+      else
+        line="None"
+        echo "$line"
+        return 1
+      fi
+    else
+      line="None"
+      echo "$line"
+      return 1
+    fi
+  }
+
+  # This is the pre-install work for a kohya installation on a runpod
+  if [ "$RUNPOD" = true ]; then
+    if [ -d "$VENV_DIR" ]; then
+      echo "Pre-existing installation on a runpod detected."
+      export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:"$VENV_DIR"/lib/python3.10/site-packages/tensorrt/
+      export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:"$VENV_DIR"/lib/python3.10/site-packages/nvidia/cuda_runtime/lib/
+      cd "$DIR" || exit 1
+      sed -i "s/interface.launch(\*\*launch_kwargs)/interface.launch(\*\*launch_kwargs,share=True)/g" ./kohya_gui.py
+    else
+      echo "Clean installation on a runpod detected."
+      cd "$BASE_DIR" || exit 1
+      git clone "$GIT_REPO"
```

Comment:
I haven't tested it yet, but this will most likely fail. 
```bash
$ DIR="/workspace/kohya_ss";
$ BASE_DIR="$(echo "$DIR" | cut -d "/" -f2)";
$ VENV_DIR="$DIR/venv";
$ echo "D: $DIR, BD: $BASE_DIR, VD: $VENV_DIR"
D: /workspace/kohya_ss, BD: workspace, VD: /workspace/kohya_ss/venv
```
on runpod we start in `/workspace/stable-diffusion/` usually (depends on template). So `$BASE_DIR` needs to be absolute path.
We can skip cloning repo, altogether since if we are running this script we most likely already are inside one. Maybe we can check if `/workspace/kohya_ss` exists and if not then clone there since we depend on this path. Or a simple solution, if we are in git repo, then `DIR="$(pwd)"`, `VENV_DIR="$DIR/venv"`

Reply:
Agree with your DIR point. I'll add some more logic to better detect and make fewer assumptions. As it stands it really only works if you're one folder deep.